### ISSUE TITLE
$argv of postexec function is not a list

### ISSUE THREAD

**Issue by GregDMeyer** [fish-shell/fish-shell#6935]
#### Issue

When I define a function to run on fish_postexec, like this:

```fish
function test_postexec --on-event fish_postexec
    echo $argv
    echo (count $argv)
end
```
the first echo prints whatever command I ran with its arguments, but the second always returns 1. For example, with that postexec function defined:

```
$ echo this command had five arguments
this command had five arguments
echo this command had five arguments
1
```

It feels like it would be more appropriate if this variable were a list containing the command and each of its arguments (though I guess that is somewhat of a subjective question).

#### System info

$ fish --version
fish, version 3.1.0

Manjaro Linux, Gnome terminal

Tried with `sh -c 'env HOME=$(mktemp -d) fish'`, same behavior

**Comment by ridiculousfish** [comment]
To be sure the user, a single command line may contain more than one command, or have blocks, etc. So I interpret this as asking for the command line to be tokenized.

I think the reason to not tokenize the command is that it the original commandline could not be recovered. For example, there would be no way of knowing if the command started with a space or not. For that reason I think it's better to send the untokenized text.

I can see adding a `string tokenize` to deal with this case though; please open a new issue if that's where you'd like to take this. Thanks for filing!

**Comment by zanchey** [comment]
fish has `read --tokenize`, which was implemented instead of `string tokenize` - see #3823.

**Comment by GregDMeyer** [comment]
thanks for your feedback! I can certainly accomplish my scripting goals without it, I was just a bit surprised by that behavior. but hearing that reasoning it makes sense.
